api/segments/migrations/0021_create_whitelisted_segments.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.2.24 on 2024-02-12 19:48
+from django.apps.registry import Apps
+from django.db import migrations, models
+import django.db.models.deletion
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.conf import settings
+from django.db import connection
+
+
+
+def create_whitelisted_segments(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    model_class = apps.get_model("segments", "WhitelistedSegment")
+    sql = f"""
+    SELECT s.id
+    FROM segments_segment s
+    LEFT OUTER JOIN segments_segmentrule sr1 ON s.id = sr1.segment_id
+    LEFT OUTER JOIN segments_segmentrule sr2 ON sr1.id = sr2.rule_id
+    LEFT OUTER JOIN segments_condition sc ON sr2.id = sc.rule_id
+    GROUP BY s.id
+    HAVING COUNT(*) > {settings.SEGMENT_RULES_CONDITIONS_LIMIT}
+    ORDER BY COUNT(*) DESC;
+    """
+
+    whitelisted_segments = []
+    with connection.cursor() as cursor:
+        cursor.execute(sql)
+        results = cursor.fetchall()
+        for result in results:
+            segment_id = result[0]
+            whitelisted_segments.append(model_class(segment_id=segment_id))
+
+    model_class.objects.bulk_create(whitelisted_segments)
```

Comment:
They're painful to write, but I'd love to see a migration test written for this.

Reply:
Ok this is now accomplished.